src/Compilers/CSharp/Test/Semantic/Semantics/InitOnlyMemberTests.cs

```diff
@@ -0,0 +1,2255 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Symbols.Metadata.PE;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.Semantics
+{
+    [CompilerTrait(CompilerFeature.InitOnlySetters)]
+    public class InitOnlyMemberTests : CompilingTestBase
+    {
+        // Spec: https://github.com/dotnet/csharplang/blob/master/proposals/init.md
+
+        // PROTOTYPE(init-only): test allowed from 'with' expression
+        // PROTOTYPE(init-only): public API, confirm behavior of IsReadOnly and IsInitOnly
+
+        // PROTOTYPE(init-only): open issues:
+        // PROTOTYPE(init-only): queue discussion on init methods (`init void Init()`) and collection initializers (`init void Add()`)
+
+        // PROTOTYPE(init-only): test dynamic scenario
+        // PROTOTYPE(init-only): test whether reflection use property despite modreq?
+        // PROTOTYPE(init-only): test behavior of old compiler with modreq. For example VB
+        // PROTOTYPE(init-only): test with ambiguous IsExternalInit types
+
+        [Fact]
+        public void TestCSharp8()
+        {
+            string source = @"
+public class C
+{
+    public string Property { get; init; }
+}
+";
+            var comp = CreateCompilation(new[] { source, IsExternalInitTypeDefinition }, parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics(
+                // (4,35): error CS8652: The feature 'init-only setters' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     public string Property { get; init; }
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "init").WithArguments("init-only setters").WithLocation(4, 35)
+                );
+        }
+
+        [Fact]
+        public void TestInitNotModifier()
+        {
+            string source = @"
+public class C
+{
+    public string Property { get; init set; }
```

Comment:
>public string Property { get; init set; } [](start = 4, length = 41)

It might be interesting to have a parsing test for this scenario. I assume there will be three accessors in the syntax tree.